scapy/layers/ipsec.py

```diff
@@ -818,6 +1082,8 @@ def __init__(self, proto, spi, seq_num=1, crypt_algo=None, crypt_key=None,
         :param crypt_key: the encryption key (only used with ESP)
         :param auth_algo: the integrity algorithm name
         :param auth_key: the integrity key
+        :param aead_algo: the aead algorithm name
```

Comment:
Couldn't `AEAD` algos be merged with `AUTH` algos? This would would allow to keep a more consistent API. (`crypt_key` vs `aead_key` ?)

If there's a good reason to keep them separate that's fine too but could you explain it? Thanks

Reply:
Hello,
Regarding IPSec processing, there are two modes: AH and ESP.
AH is Authentication Header which leads to authenticate the IP packet with an authentication algorithm.
ESP is Encrypted Secure Payload. It means that Packet will be encrypted and authenticated.

For ESP processing, you have a 2 ways to do the processing:

- Use a dedicated cipher algo with a dedicated authentication algo. Scapy implemented that.

- Use a combo algo (algo which do both encryption and authentication). Combo algo are referenced as AEAD algos (Authenticated Encryption with Associated Data). So I push this proposal to implement that.

I give you an example: CHACHA20 (encryption) and POLY1305 (authentication): In this case, AEAD processing (using AEAD CHACHA20_POLY1305 algo) is not equivalent to use CHACHA20 cipher and POLY1305 algo. Because AEAD algo (RFC) defines a one time key creation (OTK) for POLY1305 by CHACHA20.

That is why is makes no sense to "merge" AEAD algo class with Auth class or Crypt class. Even if there are functions which are common.

Reply:
Regarding IPSec processing, we should keep Crypt, Auth and AEAD algo separated.